crates/transaction-emitter-lib/src/emitter/transaction_executor.rs

```diff
@@ -99,20 +99,28 @@ impl RestApiReliableTransactionSubmitter {
                 }
             }
 
-            if result.is_ok() {
-                counters
-                    .successes
-                    .fetch_add(1, std::sync::atomic::Ordering::Relaxed);
-                if !counters.by_client.is_empty() {
+            match result {
+                Ok(()) => {
                     counters
-                        .by_client
-                        .get(&rest_client.path_prefix_string())
-                        .map(|(successes, _, _)| {
-                            successes.fetch_add(1, std::sync::atomic::Ordering::Relaxed)
-                        });
-                }
-                return Ok(());
-            };
+                        .successes
+                        .fetch_add(1, std::sync::atomic::Ordering::Relaxed);
+                    if !counters.by_client.is_empty() {
+                        counters
+                            .by_client
+                            .get(&rest_client.path_prefix_string())
+                            .map(|(successes, _, _)| {
+                                successes.fetch_add(1, std::sync::atomic::Ordering::Relaxed)
+                            });
+                    }
+                    return Ok(());
+                },
+                Err(err) => {
+                    // TODO: we should have a better way to decide if a failure is retryable
+                    if format!("{}", err).contains("SEQUENCE_NUMBER_TOO_OLD") {
```

Comment:
I see, so this and the change below. makes sense.

do you want to revert the rest of the PR, so we can land it?

Reply:
k on it